x-pack/plugins/endpoint/common/generate_data.ts

```diff
@@ -0,0 +1,410 @@
+  const rand = randomN(lenghtLimit) + 1;
+  return [...Array(rand).keys()].map(generator);
+}
+
+function randomMac(): string {
+  return [...randomNGenerator(255, 6)].map(x => x.toString(16)).join('-');
+}
+
+function randomIP(): string {
+  return [10, ...randomNGenerator(255, 3)].map(x => x.toString()).join('.');
+}
+
+function randomVersion(): string {
+  return [6, ...randomNGenerator(10, 2)].map(x => x.toString()).join('.');
+}
+
+function randomChoice<T>(arg: T[]): T {
+  return arg[randomN(arg.length)];
+}
+
+function randomString(length: number): string {
+  return [...randomNGenerator(36, length)].map(x => x.toString(36)).join('');
+}
+
+function randomHostname(): string {
+  return `Host-${randomString(10)}`;
+}
+
+export function generateRelatedEvents(
+  node: EndpointEvent,
+  generator: EndpointDocGenerator,
+  numRelatedEvents = 10
+): EndpointEvent[] {
+  const ts = node['@timestamp'] + 1000;
+  const relatedEvents: EndpointEvent[] = [];
+  for (let i = 0; i < numRelatedEvents; i++) {
+    relatedEvents.push(
+      generator.generateEvent(
+        ts,
+        node.process.entity_id,
+        node.process.parent?.entity_id,
+        randomChoice(OTHER_EVENT_CATEGORIES)
+      )
+    );
+  }
+  return relatedEvents;
+}
+
+export function generateResolverTree(
+  root: EndpointEvent,
+  generator: EndpointDocGenerator,
+  generations = 2,
+  maxChildrenPerNode = 2,
+  relatedEventsPerNode = 3,
+  percentNodesWithRelated = 100,
+  percentChildrenTerminated = 100
+): EndpointEvent[] {
+  let events: EndpointEvent[] = [root];
+  let parents = [root];
+  let timestamp = root['@timestamp'];
+  for (let i = 0; i < generations; i++) {
+    const newParents: EndpointEvent[] = [];
+    parents.forEach(element => {
+      // const numChildren = randomN(maxChildrenPerNode);
+      const numChildren = maxChildrenPerNode;
+      for (let j = 0; j < numChildren; j++) {
+        timestamp = timestamp + 1000;
+        const child = generator.generateEvent(timestamp, undefined, element.process.entity_id);
+        newParents.push(child);
+      }
+    });
+    events = events.concat(newParents);
+    parents = newParents;
+  }
+  const terminationEvents: EndpointEvent[] = [];
+  let relatedEvents: EndpointEvent[] = [];
+  events.forEach(element => {
+    if (randomN(100) < percentChildrenTerminated) {
+      timestamp = timestamp + 1000;
+      terminationEvents.push(
+        generator.generateEvent(
+          timestamp,
+          element.process.entity_id,
+          element.process.parent?.entity_id,
+          'process',
+          'end'
+        )
+      );
+    }
+    if (randomN(100) < percentNodesWithRelated) {
+      relatedEvents = relatedEvents.concat(
+        generateRelatedEvents(element, generator, relatedEventsPerNode)
+      );
+    }
+  });
+  events = events.concat(terminationEvents);
+  events = events.concat(relatedEvents);
+  return events;
+}
+
+export function generateEventAncestry(
```

Comment:
It might be helpful to add `Alert` to the function name so it's more clear that this is specific to generating Alert ancestry.